pingu/corpus/assets/label.py

```diff
@@ -54,11 +77,14 @@ class LabelList(object):
         >>> ])
     """
 
-    __slots__ = ['labels', 'idx']
+    __slots__ = ['idx', 'labels', 'utterance']
 
     def __init__(self, idx='default', labels=[]):
         self.idx = idx
-        self.labels = list(labels)
+        self.utterance = None
+
+        self.labels = []
+        self.extend(labels)
```

Comment:
`labels=None` in the argument list and `self.labels = labels or []` should save us a loop over the list of labels.

Reply:
Considering that the other constructors (`Utterance`, …) automatically set the references to the owned objects during construction, this should probably happen here too.

Reply:
Relation is set in ``append``. That is why i do all additions, by calling ``append()``.